hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -868,8 +879,71 @@ protected void initializeZKBasedSystemTrackers()
 
   // Will be overriden in test to inject customized AssignmentManager
   @VisibleForTesting
-  protected AssignmentManager createAssignmentManager(MasterServices master) {
-    return new AssignmentManager(master);
+  protected AssignmentManager createAssignmentManager(MasterServices master,
+    MasterRegion masterRegion) {
+    return new AssignmentManager(master, masterRegion);
+  }
+
+  /**
+   * Load the meta region state from the meta region server ZNode.
+   * @param zkw reference to the {@link ZKWatcher} which also contains configuration and operation
+   * @param replicaId the ID of the replica
+   * @return regionstate
+   * @throws KeeperException if a ZooKeeper operation fails
+   */
+  private static RegionState getMetaRegionState(ZKWatcher zkw, int replicaId)
```

Comment:
Region State is in zk?

Reply:
Perhaps this is old state? If so, maybe say so in doc?

The @return regionstate can be removed.

Reply:
For migrating the meta location from zk to the new local root table.